src/lib/operators/join_hash.cpp

```diff
@@ -72,43 +72,50 @@ std::shared_ptr<AbstractOperator> JoinHash::_on_deep_copy(
 void JoinHash::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {}
 
 template <typename T>
-size_t JoinHash::calculate_radix_bits(const size_t build_relation_size, const size_t probe_relation_size) {
+size_t JoinHash::calculate_radix_bits(const size_t build_side_size, const size_t probe_side_size,
+                                      const JoinMode mode) {
   /*
-    Setting number of bits for radix clustering:
-    The number of bits is used to create probe partitions with a size that can
-    be expected to fit into the L2 cache.
-    This should incorporate hardware knowledge, once available in Hyrise.
-    As of now, we assume a L2 cache size of 1024 KB (L2 cache size of recent
-    Intel Xeon CPUs), of which we use 50%.
+    The number of radix bits is used to determine the number of build partitions. The idea is to size the partitions in
+    a way that keeps the whole hash map cache resident. We aim for the largest unshared cache (for most Intel systems
+    that's the L2 cache, for Apple's M1 the L1 cache). This calculation should include hardware knowledge, once
+    available in Hyrise. As of now, we assume a cache size of 1024 KB, of which we use 75 %.
+
     We estimate the size the following way:
       - we assume each key appears once (that is an overestimation space-wise, but we
-      aim rather for a hash map that is slightly smaller than L2 than slightly larger)
-      - each entry in the hash map is a uint32_t offset (see hash_join_steps.hpp)
+        aim rather for a hash map that is slightly smaller than the cache than slightly larger)
+      - each entry in the hash map is a pair of the actual hash key and the SmallPosList storing uint32_t offsets (see
+        hash_join_steps.hpp)
   */
-  if (build_relation_size > probe_relation_size) {
+  if (build_side_size > probe_side_size) {
     /*
-      Hash joins perform best when the build relation is small. In case the
-      optimizer selects the hash join due to such a situation, but neglects that the
-      input will be switched (e.g., due to the join mode), the user will be warned.
+      Hash joins perform best when the build side is small. For inner joins, we can simply select the smaller input
+      table as the build side. For other joins, such as semi or outer joins, the build side is fixed. In this case,
+      other join operators might be more efficient. We emit performance warning in this case. Note, as of now we do not
+      consider such cases of inefficient hash joins and do not switch to other join algorithms.
```

Comment:
"As of now" does not really add any information. Maybe "In the future, the optimizer could identify these cases of potentially inefficient hash joins and switch to other join algorithms"?